src/cargo/util/toml/mod.rs

```diff
@@ -1342,7 +1342,22 @@ impl TomlManifest {
             .unwrap()
             .clone();
         package.workspace = None;
-        package.resolver = ws.resolve_behavior().to_manifest();
+        let current_resolver = package
+            .resolver
+            .as_ref()
+            .map(|r| ResolveBehavior::from_manifest(r))
+            .unwrap_or_else(|| {
+                package
+                    .edition
+                    .as_ref()
+                    .and_then(|e| e.as_defined())
+                    .map(|e| Edition::from_str(e))
+                    .transpose()
+                    .map(|e| e.unwrap_or(Edition::Edition2015).default_resolve_behavior())
+            })?;
+        if ws.resolve_behavior() != current_resolver {
+            package.resolver = Some(ws.resolve_behavior().to_manifest());
+        }
```

Comment:
imo the value of eliding a field in a programmatically generated file that is mostly read programmatically, and not by humans, is low enough that it doesn't outweigh the complexity of eliding it.

Reply:
Removed that again.